Index: src/use_case/view_scores/ViewScoresInteractor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package use_case.view_scores;\r\n\r\npublic class ViewScoresInteractor {\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/use_case/view_scores/ViewScoresInteractor.java b/src/use_case/view_scores/ViewScoresInteractor.java
--- a/src/use_case/view_scores/ViewScoresInteractor.java	(revision 7262c7fec632f61626a78021ee2efc034ae48613)
+++ b/src/use_case/view_scores/ViewScoresInteractor.java	(date 1700192093723)
@@ -1,4 +1,18 @@
 package use_case.view_scores;
 
-public class ViewScoresInteractor {
+public class ViewScoresInteractor implements ViewScoresInputBoundary{
+    final ViewScoresDataAccessInterface viewScoresDataAccessObject;
+    final ViewScoresOutputBoundary viewScoresPresenter;
+
+    public ViewScoresInteractor(ViewScoresDataAccessInterface viewScoresDataAccessObject, ViewScoresOutputBoundary viewScoresPresenter) {
+        this.viewScoresDataAccessObject = viewScoresDataAccessObject;
+        this.viewScoresPresenter = viewScoresPresenter;
+    }
+
+
+    @Override
+    public void execute() {
+        ViewScoresOutputData viewScoresOutputData = new ViewScoresOutputData(viewScoresDataAccessObject.getLastTenScores());
+        viewScoresPresenter.prepareScoreView(viewScoresOutputData);
+    }
 }
